tests/test-scribe.c

```diff
@@ -787,6 +790,33 @@ main (int argc, char *argv[])
               test_error,
               fixture_tear_down);
 
+  /* What happens if GPG gives up on us before we've finished feeding it the
+   * whole file?
+   */
+  TestData gpg_exits_prematurely_success = {
+      .image_path = image_path,
+      .signature_path = image_sig_path,
+      .gpg_path = "true",
+      .error_domain = G_IO_ERROR,
+      .error_code = G_IO_ERROR_BROKEN_PIPE,
+  };
+  g_test_add ("/scribe/gpg-exits-prematurely/success", Fixture, &gpg_exits_prematurely_success,
+              fixture_set_up,
+              test_error,
+              fixture_tear_down);
+
+  TestData gpg_exits_prematurely_failure = {
+      .image_path = image_path,
+      .signature_path = image_sig_path,
+      .gpg_path = "false",
+      .error_domain = G_IO_ERROR,
```

Comment:
Oh right! This is racy! If the tee task notices its pipe-to-GPG is closed before the subprocess task notices "GPG" has exited, we'll get BROKEN_PIPE. But if (as happened) on Jenkins the race goes the other way, we'll get GIS_IMAGE_ERROR_VERIFICATION_FAILED. I think I don't care which error we get, just that *some* error is reported.